src/daemon/utils_avltree.c

```diff
@@ -638,6 +638,7 @@ int c_avl_pick (c_avl_tree_t *t, void **key, void **value)
 	*value = n->value;
 
 	free_node (n);
+	--t->size;
```

Comment:
yeah, it's a bug that I was planning to report upstream. It's not a leak exactly, it just means that the avl_tree starts lying about its size.